usaspending_api/awards/models.py

```diff
@@ -310,6 +328,24 @@ def get_default_fields(path=None):
             "assistance_data"  # must match related_name in TransactionAssistance
         ]
 
+    @classmethod
+    def get_or_create(cls, **kwargs):
+        """Gets and updates, or creates, a Transaction
+
+        Transactions must be unique on Award, Awarding Agency, and Mod Number
+        """
+        transaction = cls.objects.filter(
+            award=kwargs.get('award'),
+            awarding_agency=kwargs.get('awarding_agency'),
+            modification_number=kwargs.get('modification_number')
+        ).first()
```

Comment:
To your question on the JIRA ticketk about a unique index. That sure would be preferable. Again, _in theory_, if we start with a fresh db, that should be fine, once the loaders are all fixed up to updated existing txns instead of inserting new ones.

The only issue I foresee is the existing historical data we have loaded. If none of that would violate then unique constraint, I think we should add it. Otherwise, we may have to put a story for that in the backlog.

If we're unable to add the unique constraint (which I agree is the preferable solution), what do you think about adding a descending sort here on `update_date` to ensure that we're getting the latest existing record when there are more than one? Then we'll at least know that if there are somehow multiple txn records, we're always updating the same one.